### PR TITLE
Fix: Selector on IndividualItemDrop

### PR DESCRIPTION
**Pull request by Zintixx**

### CODE REVIEW

File: Maple2.File.Parser/Xml/Table/IndividualItemDrop.cs [hunk 1]
```diff
@@ -16,7 +16,7 @@ namespace Maple2.File.Parser.Xml.Table;
 // ./data/xml/table/na/individualitemdrop_gearbox.xml
 [XmlRoot("ms2")]
 public partial class IndividualItemDropRoot {
-    [M2dFeatureLocale(Selector = "individualDropBoxID")] private IList<IndividualItemDrop> _individualDropBox;
+    [M2dFeatureLocale(Selector = "individualDropBoxID|dropGroup|item")] private IList<IndividualItemDrop> _individualDropBox;
```
Comment (kOchirasu) [inline Maple2.File.Parser/Xml/Table/IndividualItemDrop.cs:19]: do you need `item` here? That would mean there are multiple entries with the same `individualDropBoxID` and `dropGroup` but different `item`

File: Maple2.File.Parser/TableParser.cs [hunk 2]
```diff
@@ -603,123 +603,113 @@ public class TableParser {
...
+    public IEnumerable<(int Id, IEnumerable<IndividualItemDrop> ItemDrops)> ParseIndividualItemGearBox() {
         string xml = Sanitizer.RemoveEmpty(xmlReader.GetString(xmlReader.GetEntry("table/na/individualitemdrop_gearbox.xml")));
         xml = Sanitizer.SanitizeBool(xml);
         var reader = XmlReader.Create(new StringReader(xml));
         var data = individualItemDropSerializer.Deserialize(reader) as IndividualItemDropRoot;
         Debug.Assert(data != null);
-
-        foreach (IndividualItemDrop drop in data.individualDropBox) {
-            yield return (drop.individualDropBoxID, drop);
+        foreach (IGrouping<byte, IndividualItemDrop> group in data.individualDropBox.GroupBy(dropbox => dropbox.dropGroup)) {
```
Comment (kOchirasu) [inline Maple2.File.Parser/TableParser.cs]: This seems a bit wrong, you are returning the `group.Key` as `int Id` and missing `drop.individualDropBoxID` entirely. Wouldn't it be better to return a dictionary so you can preserve this grouping?

```cs
public IEnumerable<(int id, IDictionary<byte, IList<IndividualItemDrop>>)> Parse();
```

File: Maple2.File.Parser/Xml/Table/IndividualItemDrop.cs (hunk 1 above)
Comment (Zintixx) [reply Maple2.File.Parser/Xml/Table/IndividualItemDrop.cs:19]: yeah there's multiple entires with individualDropBoxID and dropGroups with the same values.

File: Maple2.File.Parser/TableParser.cs (hunk 2 above)
Comment (Zintixx) [reply Maple2.File.Parser/TableParser.cs]: oh I see. is there a simplified way to foreach loop this? havent tried doing something like that before

Comment (kOchirasu) [reply Maple2.File.Parser/TableParser.cs]: Untested but maybe something like:

```cs
Dictionary<byte, IList<IndividualItemDrop>> groups = data.individualDropBox.GroupBy(dropbox => dropbox.dropGroup)
    .ToDictionary(group => group.Key, group => group.ToList());
yield return (drop.individualDropBoxID, groups)
```

File: Maple2.File.Parser/TableParser.cs [hunk 3]
```diff
@@ -603,123 +603,117 @@ public class TableParser {
         }
     }
 
-    public IEnumerable<(int Id, IndividualItemDrop ItemDrop)> ParseIndividualItemDrop() {
+    public IEnumerable<(int id, IDictionary<byte, List<IndividualItemDrop>>)> ParseIndividualItemDrop() {
```
Comment (kOchirasu) [inline Maple2.File.Parser/TableParser.cs]: nit: capitalize `Id` here

**Pull request merged**